.github/actions/cache-build/action.yml

```diff
@@ -14,16 +14,14 @@ runs:
         cache-name: prod-build
         key-1: ${{ inputs.node_version }}-${{ hashFiles('yarn.lock') }}
         key-2: ${{ hashFiles('apps/**/**.[jt]s', 'apps/**/**.[jt]sx', 'packages/**/**.[jt]s', 'packages/**/**.[jt]sx', '!**/node_modules') }}
-        key-3: ${{ github.event.pull_request.number || github.ref }}
```

Comment:
@zomars @emrysal Did this absolutely need to be here? Will revert if so. Was just thinking that since we already use the `sha`, this might not be necessary.

Reply:
Turns out that github.sha here is the current commit on `main` and not the branch that's attempting to be merged in, causing bad caching issues and CI failures when in main.

Reply:
Fix is here. https://github.com/calcom/cal.com/pull/14932